internal/admission/defaulting_test.go

```diff
@@ -126,92 +127,129 @@ func TestFlow(t *testing.T) {
 		pkg.NamespaceValidationLabel: pkg.NamespaceValidationEnabled,
 	}}}
 
-	type args struct {
-		labels               map[string]string
-		shouldNotBeValidated bool
-	}
 	type want struct {
-		patchesCount     int
-		validationStatus string
+		shouldCallValidate bool
+		patches            []jsonpatch.JsonPatchOperation
 	}
 	tests := []struct {
-		name string
-		args args
-		want want
+		name        string
+		operation   admissionv1.Operation
+		inputLabels map[string]string
+		want        want
 	}{
 		{
-			name: "pod without label should pass with validation and set Success",
-			args: args{
-				labels:               nil,
-				shouldNotBeValidated: false,
-			},
+			name:        "update pod without label should pass with validation",
+			operation:   admissionv1.Update,
+			inputLabels: nil,
 			want: want{
-				patchesCount:     1,
-				validationStatus: pkg.ValidationStatusSuccess,
+				shouldCallValidate: true,
+				patches:            patchWithAddSuccessLabel(),
 			},
 		},
 		{
-			name: "pod with label Success should pass with validation",
-			args: args{
-				labels:               map[string]string{pkg.PodValidationLabel: pkg.ValidationStatusSuccess},
-				shouldNotBeValidated: false,
+			name:        "update pod with label Success should pass with validation",
+			operation:   admissionv1.Update,
+			inputLabels: map[string]string{pkg.PodValidationLabel: pkg.ValidationStatusSuccess},
+			want: want{
+				shouldCallValidate: true,
+				patches:            []jsonpatch.JsonPatchOperation{},
 			},
+		},
+		{
+			name:        "update pod with unknown label should pass with validation",
+			operation:   admissionv1.Update,
+			inputLabels: map[string]string{pkg.PodValidationLabel: "some-unknown-label"},
 			want: want{
-				patchesCount: 0,
+				shouldCallValidate: true,
+				patches:            patchWithReplaceSuccessLabel(),
 			},
 		},
 		{
-			name: "pod with label Failed should pass without validation and unchanged",
-			args: args{
-				labels:               map[string]string{pkg.PodValidationLabel: pkg.ValidationStatusFailed},
-				shouldNotBeValidated: true,
+			name:        "update pod with label Failed should pass without validation",
+			operation:   admissionv1.Update,
+			inputLabels: map[string]string{pkg.PodValidationLabel: pkg.ValidationStatusFailed},
+			want: want{
+				shouldCallValidate: false,
+				patches:            []jsonpatch.JsonPatchOperation(nil),
```

Comment:
is `[]jsonpatch.JsonPatchOperation{}`different?

Reply:
Different response patches are produced by those 2 functions even when nothing to do
- admission.Allowed("validation is not needed for pod")
- admission.PatchResponseFromRaw(req.Object.Raw, fBytes)